src/Microsoft.AspNet.OData/Formatter/Serialization/ODataSerializerContext.cs

```diff
@@ -43,8 +43,24 @@ public UrlHelper Url
             set
             {
                 _urlHelper = value;
-                InternalUrl = _urlHelper != null ? new WebApiUrlHelper(_urlHelper) : null;
+                InternalUrlHelper = _urlHelper != null ? new WebApiUrlHelper(_urlHelper) : null;
             }
         }
+
+        /// <summary>
+        /// Copy the properties this instance of <see cref="ODataSerializerContext"/> from an existing instance.
+        /// </summary>
+        /// <param name="context"></param>
+        private void CopyProperties(ODataSerializerContext context)
```

Comment:
>CopyProperties [](start = 21, length = 14)

Should this not copy InternalRequest, InternalUrlHelper, NavigationSource, SelectExpandClause, ExpandedResource, or EdmProperty, ?

Reply:
Okay; I see where this was copied from the original code didn't copy these additional properties, but should it?

---
In reply to: [152154121](https://github.com/OData/WebApi/pull/1128#discussion_r152154121) [](ancestors = 152154121)

Reply:
Setting Request and Url uses the setters so InternalRequest and InternalUrlHelper will be updated.